### PR TITLE
Use Net::IMAP.format_date instead of to_imap_date

### PR DESCRIPTION
**Pull request by tcaddy**
Use `Net::IMAP.format_date` instead of `to_imap_date`.

See https://github.com/gmailgem/gmail/pull/244#issuecomment-405390645

### CODE REVIEW

**Comment by johnnyshields** [comment]
Cheers, thank you!

**Comment by mnohai-mdsol** [comment]
this is broken, 
```NoMethodError:
  undefined method `to_imap_date' for Thu, 19 Jul 2018:Date
  Did you mean?  to_date```

File: lib/gmail/mailbox.rb [hunk 1]
```diff
@@ -32,8 +32,8 @@ def fetch_uids(*args)
         search = MAILBOX_ALIASES[args.shift].dup
         opts = args.first.is_a?(Hash) ? args.first : {}
 
-        opts[:after]      and search.concat ['SINCE', opts[:after].to_imap_date]
-        opts[:before]     and search.concat ['BEFORE', opts[:before].to_imap_date]
+        opts[:after]      and search.concat ['SINCE', Net::IMAP.format_date(opts[:after])]
+        opts[:before]     and search.concat ['BEFORE', Net::IMAP.format_date(opts[:before])]
         opts[:on]         and search.concat ['ON', opts[:on].to_imap_date]
```
Comment (mnohai-mdsol) [inline lib/gmail/mailbox.rb]: forgot to remove the ```to_imap_date``` call here

Comment (johnnyshields) [reply lib/gmail/mailbox.rb]: fixing now

Comment (johnnyshields) [reply lib/gmail/mailbox.rb:37]: done.

Comment (mjnohai) [reply lib/gmail/mailbox.rb:37]: thanks for the fix, everything is working gain. Cheers! 👍

**Pull request merged**